lib/utils.js

```diff
@@ -0,0 +1,154 @@
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+'use strict';
+
+const defaultCaveatRegistry = require('./caveatRegistry');
+const vocab = require('./vocab');
+
+const api = {};
+module.exports = api;
+
+const {
+  capabilityDelegationUri,
+  caveatUri,
+  invocationTargetUri,
+  invokerUri,
+  parentCapabilityUri
+} = vocab;
+
+/**
+ * Retrieves a capability's parent's invoker
+ *
+ * @param {Object} capability - the fully expanded JSON-LD document for the
+ *                              object capability
+ * @param {Object} documentLoader - a JSON-LD document loader
+ *
+ */
+api.getParentInvoker = async ({capability, documentLoader}) => {
+  const {document: parentCapability} =
+    await documentLoader(capability[parentCapabilityUri]);
+
+  return parentCapability.invoker;
+};
+
+/**
+ * Plucks the single element of an array that contains a single item.
+ *
+ * @param {Array} array - an array with one item.
+ */
+api.getOneOrDie = array => {
+  if (!Array.isArray(array) || array.length != 1) {
+    throw new Error('Expected an array of size 1');
+  }
+  return array[0];
+};
+
+/**
+ * Retrieves the invoker from a capability.
+ *
+ * @param {Object} capability - the fully expanded JSON-LD document for the
+ *                              object capability
+ */
+api.getInvoker = capability => {
+  const invoker = capability[invokerUri];
+  const id = capability['@id'];
+  if(!invoker || !id) {
+    throw new Error('Invoker not found for capability');
+  }
+
+  return invoker || id;
+};
+
+/**
+ * Verifies the caveats for a given object capability.
+ *
+ * @param {Object} capability - the fully expanded JSON-LD document for the
+ *                              object capability
+ * @param {Object} jsigs - a configured instance of jsonld-signatures
+ * @param {Object} proofPurposeOptions  - a set of options for validating the
+ *                                        proof purpose.
+ */
+api.verifyCapability = async ({capability, jsigs, proofPurposeOptions}) => {
+  // Capability does not point to a parent capability and it was capable of
+  // delegating a capability
+  if(!(parentCapabilityUri in capability) &&
+    capabilityDelegationUri in capability) {
+    const {expectedTarget} = proofPurposeOptions;
+    // target is the root capability unless an invocation target is specified
+    const target = invocationTargetUri in capability ?
+      api.getOneOrDie(capability[invocationTargetUri]) : capability['@id'];
+    // It's the toplevel capability, which means it's valid if the target
+    // matches the expected target
+    return target === expectedTarget;
+  } else {
+    // Otherwise, we have to check the signature
+    return jsigs.verify(capability, {
+      proofPurpose: 'CapabilityDelegation',
+      proofPurposeOptions: proofPurposeOptions
```

Comment:
just `proofPurposeOptions`.

Reply:
Fix https://github.com/digitalbazaar/ocapld.js/pull/1/commits/4e44f6d9e1850dc8f88ad995e0ceb2d076888b71

Reply:
I'm not sure what you mean @mattcollier 

But I do see an error; this shouldn't be passing in the proofPurpose as a string, it should be passing in the uninstantiated class... that's an error

Reply:
Oops, didn't mean to mark that as a review